test/model/message_test.dart

```diff
@@ -919,4 +919,23 @@ void main() {
       });
     });
   });
+
+  test('disposing multiple registered MessageListView instances', () async {
+    // Regression test for: https://github.com/zulip/zulip-flutter/issues/810
+    await prepare(narrow: const MentionsNarrow());
+    MessageListView.init(store: store, narrow: const StarredMessagesNarrow());
+
+    Condition<Object?> conditionMessageListView(Narrow narrow) =>
+      (it) => it.isA<MessageListView>().narrow.equals(narrow);
+
+    check(store.debugMessageStore.debugMessageLists).deepEquals([
+      conditionMessageListView(const MentionsNarrow()),
+      conditionMessageListView(const StarredMessagesNarrow()),
+    ]);
```

Comment:
My first thought here is that this is probably clearer if `conditionMessageListView` just gets inlined at its two call sites. Less abstraction to work through, and not really any longer.

But I think this can actually go a step further and just check the length of the list of [MessageListView]s, as you do in the other test. The fact that it's these two particular narrows is sort of inevitable given the setup of the test, and it's not what this test is intended to check on.

(Also conceivably it could iterate in the other order. The current implementation doesn't do that, because the set comes from a set literal `{}` and that's a `LinkedHashSet`, which preserves the insertion order; but that's not a guarantee that `MessageStoreImpl` otherwise has any reason to make, and if it didn't then this test would sometimes break.)